tools/config_filter.py

```diff
@@ -0,0 +1,84 @@
+################################################################################
+# Copyright (C) 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+# SPDX-License-Identifier: Apache-2.0
+################################################################################
+
+import math
+from abc import ABC, abstractmethod
+from typing import List
+
+import mmh3
+
+
+class ConfigFilter(ABC):
+
+    @abstractmethod
+    def already_met(self, v: List[int]) -> bool:
+        pass
+
+    @abstractmethod
+    def add(self, v: List[int]):
+        pass
+
+    @abstractmethod
+    def save(self):
+        pass
+
+    def load(self, data):
+        pass
+
+
+class BloomFilter(ConfigFilter):
```

Comment:
What are these for?

Reply:
They are for the tuner. I guess we don't really need this currently. `ConfigFilter` will check if a config (generated from tuner) is already met before and skip it. `BloomFilter` uses O(1) space to check the existence of "unlimited“ number elements. We can remove this for now..